consensus/src/lib.rs

```diff
@@ -173,13 +185,15 @@ impl Consensus {
 impl Consensus {
     async fn read_state(&self) -> Result<State, Error> {
         let raw_state = self.state_storage.read_file(STATE_FILE_NAME).await?;
-        let state: State = serde_spb::from_str(&raw_state)?;
+        let state: State = serde_spb::from_slice(&hex::decode(raw_state)?)?;
         Ok(state)
     }
 
     async fn commit_state(&mut self, state: &State) -> Result<(), Error> {
+        // We can't use json because of a non-string map
+        let data = hex::encode(serde_spb::to_vec(state).unwrap());
```

Comment:
Does this mean that State was stored as json and now stored as a byte slice?
I don't understand reason why we cannot use json.

Reply:
Json supports only string as the key of a dictionary